### PR TITLE
Feature/build failure

### PR DESCRIPTION
**Pull request by abunur**
Using an approach[ like this ](https://anshsachdevawork.medium.com/securing-your-keys-with-gradle-variables-648b7faecf67)might be better. This way, we don't need to store environment variables in the **gradle.properties** file. By committing **gradle.properties**, we allow the project to build without having to alert the user to make changes that would normally be set by Android Studio. Instead, we alert them to creating their own environment variables in their own files that have nothing to do with the gradle files, and are just referenced by the build.grade

More info on the difference in approach can be found [here](https://discuss.gradle.org/t/best-practice-for-gradle-properties-and-scm/5809) - see the discussion.

Overall I think it's a better experience for our customers if we keep **gradle.properties** under version control.

This PR updates the readme, removes environment variables from **gradle.properties**, and puts them instead into ignored files in the **secrets** directory at the project root.

Once this is merged, we will need to also update[ this Notion document](https://www.notion.so/getditto/Environment-Variables-78261e05a2b44a299ee388f06e9ff86a?pvs=4)

### CODE REVIEW

File: Android/app/build.gradle [hunk 1]
```diff
@@ -36,9 +36,23 @@ android {
     }
 
     buildTypes {
+        debug{
+            def credsFile = rootProject.file("secure\\debug_creds.properties")
```
Comment (phatblat) [inline Android/app/build.gradle:40]: Does this backslash work on non-Windows systems?

Comment (abunur) [reply Android/app/build.gradle:40]: > Does this backslash work on non-Windows systems?

It works on Mac, I can't vouch for anything else 😅

**Comment by phatblat** [comment]
This approach is great. Gradle properties aren't secure and the project properties are typically shared. I often cheat and use the user-level `~/.gradle/gradle.properties` to override, but this behavior isn't known by everyone.

> Project-level ‘gradle.properties’ files should always be under source control, period. User-specific properties belong into ‘~/.gradle/gradle.properties’, and override project-level properties.

This is a good approach and is _explicit_ as opposed to the gradle user properties which are _implicit_.

File: Android/app/build.gradle (hunk 1 above)
Comment (phatblat) [reply Android/app/build.gradle:40]: If it works on Macs, that's good enough for me.

File: Android/app/build.gradle [hunk 2]
```diff
@@ -36,9 +36,23 @@ android {
     }
 
     buildTypes {
+        debug{
```
Comment (phatblat) [inline Android/app/build.gradle:39]: Nit: spacing

```suggestion
        debug {
```

File: README.md [hunk 3]
```diff
@@ -39,6 +39,21 @@ Powered by [Ditto](https://ditto.live/).
...
+You need to setup some environment variables in order to build this project:
+
+1. In your project root, create a directory called **secret**
+2. Add two files to that directory called **debug_creds.properties** and **release_creds.properties**, for the debug and release build variants as defined in the app **build.gradle** file.
+2. Add the following environment variables to each credential file, substituting your own values:
+```
+    # Environment Variables  
+    DITTO_APP_ID = "replace with your app id"  
+    DITTO_AUTH_TOKEN = "replace with your auth token"  
+    DITTO_AUTH_PROVIDER = "replace with your auth provider"
```
Comment (phatblat) [inline README.md:54]: Values in `.properties` files shouldn't be quoted. Values are all `String` and any leading (after the `=`) or trailing space in the value is dropped automatically, but quotes end up in the resulting `String`. I don't know how leading spaces in the key are handled, so I always have them without any leading space.

```suggestion
DITTO_APP_ID = replace with your app id
DITTO_AUTH_TOKEN = replace with your auth token
DITTO_AUTH_PROVIDER = replace with your auth provider
```

**Review by phatblat — CHANGES_REQUESTED** [review]
Example gradle properties need an edit.

**Review by kndoshn — COMMENTED** [review]
This is great. Thank you!

We need to do the same thing to [demoapp-inventory](https://github.com/getditto/demoapp-inventory) too.
- https://github.com/getditto/demoapp-inventory/issues/22

**Review by kndoshn — COMMENTED** [review]
I was hoping that the Android app also can refer to the `.env` values as the iOS does, but I didn't find the way.
`.env` is placed at the root directory, and it will be simpler if we could use it from the Android since the values are the same.

**Review by kndoshn — COMMENTED** [review]
Will this PR allow these lines in `defaultConfig` to be removed since you added similar things in `buildTypes`?

https://github.com/getditto/demoapp-skyservice/blob/a50d8c9b307ccaa79736ceb875923de9213c01ba/Android/app/build.gradle#L17-L25

**Comment by phatblat** [comment]
Good callout, @kndoshn. The gradle properties used in `defaultConfig` to set build config values are redundant now with this change.

**Comment by okdistribute** [comment]
Where are we on this, can we merge it?

**Comment by abunur** [comment]
> Where are we on this, can we merge it?
LOL I'm literally looking at this right now.
great minds think alike? 😜

**Comment by abunur** [comment]
@kndoshn @okdistribute this branch / PR has been updated, just needs approval for merge

**Review by okdistribute — APPROVED** [review]

**Comment by abunur** [comment]
@okdistribute @phatblat thanks for the additional commits, it won't let me auto merge though so I'm overriding it to merge the PR, hope that's ok. I think b/c the commits were by other than me, but I can't approve changes on my own PR according to Github

**Comment by phatblat** [comment]
Sorry @abunur. It was probably my "request for changes" which I should have cleared by approving. I got distracted with the tangent our convo took us in.

**Comment by abunur** [comment]
> Sorry @abunur. It was probably my "request for changes" which I should have cleared by approving. I got distracted with the tangent our convo took us in.

no worries, I think it's my fault for not making the changes you requested, and they were super small/ simple anyway

**Pull request merged**